modules/core/src/main/scala/org/tessellation/infrastructure/snapshot/LatestBalancesStorage.scala

```diff
@@ -0,0 +1,19 @@
+package org.tessellation.infrastructure.snapshot
+
+import cats.effect.Async
+import cats.syntax.functor._
+
+import org.tessellation.domain.snapshot.GlobalSnapshotStorage
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.balance.Balance
+import org.tessellation.sdk.domain.collateral.LatestBalances
+
+object LatestBalancesStorage {
+
+  def make[F[_]: Async](globalSnapshotStorage: GlobalSnapshotStorage[F]) =
```

Comment:
if you want, you can mix it together with GlobalSnapshotStorage, but on implementation level (not in the trait)

Reply:
Do you mean doing sth like this:
 ````
object GlobalSnapshotStorage {

  def make[F[_]: Async: KryoSerializer](
    globalSnapshotLocalFileSystemStorage: GlobalSnapshotLocalFileSystemStorage[F],
    inMemoryCapacity: NonNegLong
  ): F[GlobalSnapshotStorage[F] with LatestBalances[F]] = {
````
and then 
```
`object Storages {

  def make[F[_]: Async: KryoSerializer](
    sdkStorages: SdkStorages[F],
    snapshotConfig: SnapshotConfig
  ): F[Storages[F]] =
    for {
     [...]
      globalSnapshotStorage <- GlobalSnapshotStorage
        .make[F](globalSnapshotLocalFileSystemStorage, snapshotConfig.inMemoryCapacity)
      latestBalancesStorage = globalSnapshotStorage`
```
Or sth like this:
```
`object Services {

  def make[F[_]: Async: Random: KryoSerializer: SecurityProvider](
   [...]
  ): F[Services[F]] =
    for {
      metrics <- Metrics.make[F]
      consensus <- GlobalSnapshotConsensus
        .make[F](
          [...]
        )
      latestBalancesStorage = new LatestBalances {
        def getLatestBalances = storages.globalSnapshot.head.map(_.map(_.info.balances))
      }
      collateralService = Collateral.make[F](cfg.collateral,latestBalancesStorage)
```

Reply:
the first approach but then:

```scala
object Storages {

  def make[F[_]: Async: KryoSerializer](
    sdkStorages: SdkStorages[F],
    snapshotConfig: SnapshotConfig
  ): F[Storages[F]] =
    for {
     [...]
      globalSnapshotStorage <- GlobalSnapshotStorage
        .make[F](globalSnapshotLocalFileSystemStorage, snapshotConfig.inMemoryCapacity)
```

without

```scala
      latestBalancesStorage = globalSnapshotStorage
```
      
as globalSnapshotStorage already satisfies the `LatestBalances` trait, so this will work:

```scala
class Collateral[F[_]](latestBalances: LatestBalances[F]) { ... }

[...]

globalSnapshotStorage <- GlobalSnapshotStorage.make[F](...)
collateral = Collateral.make[F](cfg.collateral, globalSnapshotStorage)
```

Reply:
But in this case I would also have to update `Storages` like this:
```scala
sealed abstract class Storages[F[_]] private (
  [...]
  val globalSnapshot: GlobalSnapshotStorage[F] with LatestBalances[F]
)
```
Is it ok?